File: test/df.jl
# /usr/bin/env julia1.6.1

""" install if needed
    using Pkg; Pkg.add("CSV"); Pkg.add("DataFrames"); Pkg.add("Time");
"""

using CSV
using DataFrames
using Logging


avg = 0.0
sum = 0.0
the_end = 1000
logger = SimpleLogger(open("log/test_df_jl.txt", "a+"))


function main()
    df = DataFrame(CSV.File("database/source.csv"))
    return 0
end;


for i = 0:the_end
    clock = @elapsed main()
    global sum += clock
    with_logger(logger) do
        @info(clock)
    end
end;

avg = sum/the_end
with_logger(logger) do
    @info("mean", avg)
end

File: test/df.py
# /usr/bin/env python3.9.0

import logging
import time
from pandas import read_csv as pd_read_csv


if __name__ == "__main__" :
    logging.basicConfig(filename=("log/test_df_py.txt"),
                level=logging.DEBUG, 
                format=' %(asctime)s - %(levelname)s - %(message)s')
    END = 1000 
    sum = 0
    for i in range(END):
        clock = time.time()
        try: 
            df = pd_read_csv("database/source.csv")
        except:
            logging.error("pandas did not find the dataset source")
        clock = time.time() - clock
        sum += clock
        logging.debug(clock)
    logging.info("AVG - " + str(sum/END))